policy-controller/k8s/index/src/index.rs

```diff
@@ -696,39 +610,85 @@ impl kubert::index::IndexNamespacedResource<k8s::policy::NetworkAuthentication>
     }
 }
 
-impl kubert::index::IndexNamespacedResource<k8s_gateway_api::HttpRoute> for Index {
-    fn apply(&mut self, route: k8s_gateway_api::HttpRoute) {
-        self.apply_route(route)
-    }
-
-    fn delete(&mut self, ns: String, name: String) {
-        self.delete_route(ns, name)
-    }
-
-    fn reset(
-        &mut self,
-        routes: Vec<k8s_gateway_api::HttpRoute>,
-        deleted: HashMap<String, HashSet<String>>,
-    ) {
-        self.reset_route(routes, deleted)
-    }
-}
-
 impl kubert::index::IndexNamespacedResource<k8s::policy::HttpRoute> for Index {
     fn apply(&mut self, route: k8s::policy::HttpRoute) {
-        self.apply_route(route)
+        let ns = route.namespace().expect("HttpRoute must have a namespace");
+        let name = route.name_unchecked();
+        let _span = info_span!("apply", %ns, %name).entered();
+
+        let route_binding = match route.try_into() {
+            Ok(binding) => binding,
+            Err(error) => {
+                tracing::info!(%ns, %name, %error, "Ignoring HTTPRoute");
+                return;
+            }
+        };
+
+        self.ns_or_default_with_reindex(ns, |ns| ns.policy.update_http_route(name, route_binding))
```

Comment:
I don't really follow these changes. Can we leave them as-is? We're likely to have to support gateways eventually... keeping the functions general/resusable seems fine, and helps to keep the diff minimal.

Reply:
sure, we can put these back; i just figured it wasn't worth having those functions factored out if they were only called in one place.